grapple/types/redirects.py

```diff
@@ -1,38 +1,76 @@
+import copy
+
+from typing import Optional
+
 import graphene
 
-from django.conf import settings
 from wagtail.contrib.redirects.models import Redirect
+from wagtail.models import Page, Site
+
+from grapple.types.sites import SiteObjectType
 
 from .pages import get_page_interface
 
 
-class RedirectType(graphene.ObjectType):
+class RedirectObjectType(graphene.ObjectType):
     old_path = graphene.String(required=True)
     old_url = graphene.String(required=True)
-    new_url = graphene.String(required=True)
+    new_url = graphene.String(required=False)
     page = graphene.Field(get_page_interface())
+    site = graphene.Field(SiteObjectType, required=False)
     is_permanent = graphene.Boolean(required=True)
 
-    # Give old_path with BASE_URL attached.
-    def resolve_old_url(self, info, **kwargs):
-        return settings.BASE_URL + self.old_path
+    def resolve_old_url(self, info, **kwargs) -> str:
+        """
+        Resolve the value of `old_url` using the `root_url` of the associated
+        site and `old_path`.
+        Note: `self.site` should never be none because of `resolve_redirects`.
+        """
+        if self.site:
+            return f"{self.site.root_url}/{self.old_path}"
+
+        if self.site is None:
+            return None
 
     # Get new url
-    def resolve_new_url(self, info, **kwargs):
-        if self.redirect_page is None:
+    def resolve_new_url(self, info, **kwargs) -> Optional[str]:
+        if self.redirect_page:
+            return self.redirect_page.url
+
+        if self.link:
             return self.link
 
-        return self.redirect_page.url
+        return None
 
     # Return the page that's being redirected to, if at all.
-    def resolve_page(self, info, **kwargs):
+    def resolve_page(self, info, **kwargs) -> Optional[Page]:
         if self.redirect_page is not None:
             return self.redirect_page.specific
 
+    class Meta:
+        name = "Redirect"
+
 
 class RedirectsQuery:
-    redirects = graphene.List(graphene.NonNull(RedirectType), required=True)
+    redirects = graphene.List(graphene.NonNull(RedirectObjectType), required=True)
 
     # Return all redirects.
     def resolve_redirects(self, info, **kwargs):
```

Comment:
```suggestion
    def resolve_redirects(self, info, **kwargs) -> list[Redirect]:
```

Reply:
Addressed in cba26966a00bbfb670c390908a84cd91b8cfb4c2

Reply:
Does this work locally for you when running `python manage.py test`? If so, what python version are you using?

Using Python v3.8.11, I get a `TypeError` (_TypeError: 'type' object is not subscriptable_) unless I change to:

```suggestion  
    def resolve_redirects(self, info, **kwargs) -> List[Redirect]:
```

(and importing `List` from `typing`)

Reply:
I'm using Python v3.11.2. It works locally for me, though changing it to `-> List[Redirects]` works as well.

Reply:
Went with your suggestion, since it seems to work with more Python versions. c2a53657cc16941c4ea4a6307f0e21fc27809e99